### PR TITLE
Add .spi.yml for Swift Package Index DocC support

### PR DESCRIPTION
**Pull request by yim-lee**

### CODE REVIEW

File: .spi.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+version: 1
+builder:
+  configs:
+    - documentation_targets: [Tracing]
```
Comment (slashmo) [inline .spi.yml:4]: I'm not very familiar with the Swift package index yet and its DocC support, but would this also surface the documentation from the `Instrumentation` target given that `Tracing` depends on it, or would we want to declare both as `documentation_targets` here?

Comment (yim-lee) [reply .spi.yml:4]: I guess I am not sure either. Let me include it just in case.

**Review by tomerd — APPROVED** [review]

**Comment by slashmo** [comment]
@yim-lee I just checked, with this SPI configuration when you click on "Documentation" on the SPI website it takes you to the `Instrumentation` documentation instead of `Tracing` without any UI to switch to the `Tracing` documentation. Changing the path works, but IMHO we should definitely optimize for folks seeing the `Tracing` documentation first. I don't know if it's enough to flip the order of the `documentation_targets` array to do so.

**Comment by yim-lee** [comment]
@finestructure Please see [slashmo's comment above](https://github.com/apple/swift-distributed-tracing/pull/72#issuecomment-1337035952). Is there a way for us to specify the "landing" target?

**Comment by finestructure** [comment]
That's down to the order - we're landing on the first target in the list IIRC!

**Comment by finestructure** [comment]
I can manually trigger a doc re-build out of turn to verify without waiting for the 24 cooldown, just let me know.

**Comment by yim-lee** [comment]
> That's down to the order - we're landing on the first target in the list IIRC!

Oh great! Thanks for the quick response @finestructure!

**Comment by yim-lee** [comment]
@slashmo - opened https://github.com/apple/swift-distributed-tracing/pull/74

**Comment by yim-lee** [comment]
@finestructure Can you please trigger a rebuild? Thanks in advance! 🙏

**Comment by finestructure** [comment]
That's rebuilt with revision e8e428b6db9d2b1a4675feb5dbfbd947a7b5bc1c

**Comment by finestructure** [comment]
Ok, both are there, got confused by our own UX there for a sec and couldn't find the target switcher 😅🤔

CleanShot 2022-12-05 at 19 29 16@2x

**Comment by yim-lee** [comment]
Thanks @finestructure. You are awesome!

**Comment by ktoso** [comment]
Great, thanks everyone!

**Pull request merged**